### PR TITLE
integration: update nydus tests

### PR DESCRIPTION
**Pull request by ChengyuZhu6**
To support the v0.12.0 nydus-snapshotter, we need to update the config files and the commandline to start nydus-snapshotter.

Depends-on: github.com/kata-containers/kata-containers#7961

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
This seems okay to me as long as the remaining tests pass. As I mentioned in the other PR, I think we might need these same changes in the kata-containers copy of these tests too.

**Comment by ChengyuZhu6** [comment]
> This seems okay to me as long as the remaining tests pass. As I mentioned in the other PR, I think we might need these same changes in the kata-containers copy of these tests too.

Thank you for pointing that out. I have already add these changes in https://github.com/kata-containers/kata-containers/pull/7961

**Comment by fidencio** [comment]
The current test is failing with:
```
11:23:58 Running nydus integration tests
11:23:59 Image is up to date for sha256:dcac4621b8636ac822ab7737648a272fcf378b8df6d315816036f00d5536d1e2
11:24:02 Pod 78209a269a2963096fd9537940115c8afc1727ea7002e72f4e275de55500895e created
11:24:02 Container 9a058b4e2f9855c95716684f21714f3b700e158fafe4bd0877558ee37443f49d created
11:24:03 time="2023-09-17T09:24:02Z" level=fatal msg="starting the container \"9a058b4e2f9855c95716684f21714f3b700e158fafe4bd0877558ee37443f49d\": rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: Others(\"failed to handler message create container\\n\\nCaused by:\\n    0: create\\n    1: handler rootfs\\n    2: new nydus rootfs\\n    3: failed to do rafs mount\\n    4: fail to attach rafs \\\"/var/lib/containerd-nydus/snapshots/2/fs/image/image.boot\\\"\\n    5: add share fs mount\\n    6: Mount rafs at /rafs/9a058b4e2f9855c95716684f21714f3b700e158fafe4bd0877558ee37443f49d/rootfs_lower error: Failed to Mount backend Some(\\\"rafs\\\") at /rafs/9a058b4e2f9855c95716684f21714f3b700e158fafe4bd0877558ee37443f49d/rootfs_lower mount config FsMountConfigInfo { ops: \\\"mount\\\", fstype: Some(\\\"rafs\\\"), source: Some(\\\"/var/lib/containerd-nydus/snapshots/2/fs/image/image.boot\\\"), mountpoint: \\\"/rafs/9a058b4e2f9855c95716684f21714f3b700e158fafe4bd0877558ee37443f49d/rootfs_lower\\\", config: Some(\\\"{\\\\\\\"device\\\\\\\":{\\\\\\\"backend\\\\\\\":{\\\\\\\"type\\\\\\\":\\\\\\\"registry\\\\\\\",\\\\\\\"config\\\\\\\":{\\\\\\\"readahead\\\\\\\":false,\\\\\\\"host\\\\\\\":\\\\\\\"ghcr.io\\\\\\\",\\\\\\\"repo\\\\\\\":\\\\\\\"dragonflyoss/image-service/alpine\\\\\\\",\\\\\\\"proxy\\\\\\\":{\\\\\\\"fallback\\\\\\\":false},\\\\\\\"timeout\\\\\\\":5,\\\\\\\"connect_timeout\\\\\\\":5,\\\\\\\"retry_limit\\\\\\\":2}},\\\\\\\"cache\\\\\\\":{\\\\\\\"type\\\\\\\":\\\\\\\"blobcache\\\\\\\",\\\\\\\"config\\\\\\\":{\\\\\\\"work_dir\\\\\\\":\\\\\\\"/var/lib/containerd-nydus/cache\\\\\\\",\\\\\\\"disable_indexed_map\\\\\\\":false}}},\\\\\\\"mode\\\\\\\":\\\\\\\"direct\\\\\\\",\\\\\\\"digest_validate\\\\\\\":false,\\\\\\\"enable_xattr\\\\\\\":true,\\\\\\\"fs_prefetch\\\\\\\":{\\\\\\\"enable\\\\\\\":true,\\\\\\\"prefetch_all\\\\\\\":true,\\\\\\\"threads_count\\\\\\\":8,\\\\\\\"merging_size\\\\\\\":1048576,\\\\\\\"bandwidth_rate\\\\\\\":0}}\\\"), tag: \\\"kataShared\\\", prefetch_list_path: None, dax_threshold_size_kb: None }\\n       \\n       Caused by:\\n           vmm action error: FsDevice(AttachBackendFailed(\\\"attach/detach a backend filesystem failed:: missing field `version` at line 1 column 489\\\"))\"): unknown"
11:24:03 [nydus_tests.sh:164] ERROR: sudo -E crictl start 9a058b4e2f9855c95716684f21714f3b700e158fafe4bd0877558ee37443f49d
11:24:03 Running teardown
11:24:03 integration/nydus/nydus_tests.sh: line 189: 41619 Killed                  nohup /usr/local/bin/containerd-nydus-grpc --config /etc/snapshotter-config.toml --nydusd-config /etc/nydusd-config.json --log-to-stdout > /dev/null 2>&1
11:24:03 42024
11:24:03 ERROR: Found unexpected /opt/kata/runtime-rs/bin/containerd-shim-kata-v2 present
11:24:03 
11:24:03 script:
11:24:03   name: 'integration/nydus/nydus_tests.sh'
11:24:03   pid: 41531
11:24:03   directory: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests'
11:24:03   details: 'drwxrwxr-x 15 jenkins jenkins ? 4096 Sep 17 09:23 /tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests'
11:24:03 failure:
11:24:03   function: 'check_processes'
11:24:03   file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/lib/common.bash'
11:24:03   line: 172
11:24:03   name: 'integration/nydus/nydus_tests.sh'
11:24:03 stacktrace:
11:24:03   0: {function: 'dump_details', file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/lib/error.sh', line: 171}
11:24:03   1: {function: 'die', file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/lib/common.bash', line: 70}
11:24:03   2: {function: 'check_processes', file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/lib/common.bash', line: 172, current='yes'}
11:24:03   3: {function: 'teardown', file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/integration/nydus/nydus_tests.sh', line: 206}
11:24:03   4: {function: 'run_test', file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/integration/nydus/nydus_tests.sh', line: 1}
11:24:03   5: {function: 'main', file: '/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/src/github.com/kata-containers/tests/integration/nydus/nydus_tests.sh', line: 215}
11:24:03 process-hierarchy:
11:24:03   0: {pid: 41531, command: 'bash -f integration/nydus/nydus_tests.sh', current='yes'}
11:24:03   1: {pid: 41523, command: 'make nydus'}
11:24:03   2: {pid: 41522, command: 'sudo -E PATH=/tmp/jenkins/workspace/kata-containers-dragonball-tests-ubuntu-20.04-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin CRI_RUNTIME=containerd bash -c make nydus'}
11:24:03   3: {pid: 41512, command: '/bin/bash .ci/run.sh'}
11:24:03   4: {pid: 23388, command: '/bin/bash .ci/jenkins_job_build.sh github.com/kata-containers/tests -t all'}
11:24:03   5: {pid: 23343, command: 'bash -x ci_entry_point.sh [https://github.com/kata-containers/tests.git'}](https://github.com/kata-containers/tests.git)
11:24:03   6: {pid: 23338, command: '/bin/bash /tmp/jenkins15866816435002735917.sh'}
11:24:03   7: {pid: 23205, command: 'java -jar remoting.jar -workDir /home/jenkins/remoting'}
11:24:03   8: {pid: 23201, command: 'sshd: jenkins@notty'}
11:24:03   9: {pid: 23126, command: 'sshd: jenkins [priv]'}
11:24:03   10: {pid: 1749, command: 'sshd: /usr/sbin/sshd -D [listener] 0 of 10-100 startups'}
11:24:03   11: {pid: 1, command: '/sbin/init'}
11:24:03 time: '2023-09-17T09:24:03+00:00'
11:24:03 runtime-seconds: 8
11:24:03 host:
```

**Comment by fidencio** [comment]
The Fedora tests were marked as non-required following up this thread on slack: https://katacontainers.slack.com/archives/C01HUGDR0CE/p1695043467731959

tl;dr: docker-ce.repo file coming from the official docker page is empty, it's not related to Fedora, but rather an issue on docker side.

**Review by fidencio — APPROVED** [review]
lgtm, thanks for the work @ChengyuZhu6!
